datastream-server/src/main/java/com/linkedin/datastream/server/assignment/LoadBasedPartitionAssignmentStrategyFactory.java

```diff
@@ -0,0 +1,87 @@
+/**
+ *  Copyright 2021 LinkedIn Corporation. All rights reserved.
+ *  Licensed under the BSD 2-Clause License. See the LICENSE file in the project root for license information.
+ *  See the NOTICE file in the project root for additional information regarding copyright ownership.
+ */
+package com.linkedin.datastream.server.assignment;
+
+import java.util.Optional;
+import java.util.Properties;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.linkedin.datastream.common.VerifiableProperties;
+import com.linkedin.datastream.common.zk.ZkClient;
+import com.linkedin.datastream.server.DatastreamSourceClusterResolver;
+import com.linkedin.datastream.server.DummyDatastreamSourceClusterResolver;
+import com.linkedin.datastream.server.api.strategy.AssignmentStrategy;
+import com.linkedin.datastream.server.api.strategy.AssignmentStrategyFactory;
+import com.linkedin.datastream.server.providers.NoOpPartitionThroughputProvider;
+import com.linkedin.datastream.server.providers.PartitionThroughputProvider;
+
+import static com.linkedin.datastream.server.assignment.BroadcastStrategyFactory.CFG_MAX_TASKS;
+import static com.linkedin.datastream.server.assignment.StickyMulticastStrategyFactory.CFG_IMBALANCE_THRESHOLD;
+
+
+/**
+ * A factory for creating {@link LoadBasedPartitionAssignmentStrategy} instances
+ */
+public class LoadBasedPartitionAssignmentStrategyFactory implements AssignmentStrategyFactory {
```

Comment:
Can you extend `StickyPartitionAssignmentStrategyFactory`? Most of this code is duplicate and you should be able to get rid of it.

Reply:
I addressed this by moving configs to a separate class and extending `LoadBasedPartitionAssignmentStrategy` from `StickyPartitionAssignmentStrategy`.